com/uber/kafkaSpraynozzle/stats/StatsDReporter.java

```diff
@@ -0,0 +1,61 @@
+package com.uber.kafkaSpraynozzle.stats;
+
+import com.timgroup.statsd.NonBlockingStatsDClient;
+import com.timgroup.statsd.StatsDClient;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+/**
+ * Statsd StatsReporter implementation
+ *
+ * To initialize this class you need to provide it the path to a .properties file containing:
+ * statsd.host - the statsd host
+ * statsd.port - the statsd port
+ * statsd.statsPrefix (optional) - a prefix for all stat names
+ */
+public class StatsdReporter implements StatsReporter {
+    private final StatsDClient statsd;
+    private final String statsPrefix;
+    private final String statsdHost;
+    private final Integer statsdPort;
+
+    public StatsdReporter(String propertyFilePath) throws IOException {
+        Properties statsProperties = new Properties();
+        InputStream fileInputStream;
+        try {
+            fileInputStream = new FileInputStream(propertyFilePath);
+            statsProperties.load(fileInputStream);
+            fileInputStream.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("Failed to find property file: " + propertyFilePath);
+            throw e;
+        } catch (IOException e) {
+            System.out.println("Failed to load properties: " + e.getMessage());
+            throw e;
+        }
+
+        this.statsPrefix = statsProperties.getProperty("statsd.statsPrefix", "");
+        this.statsdHost = statsProperties.getProperty("statsd.host");
+        this.statsdPort = Integer.valueOf(statsProperties.getProperty("statsd.port"));
+        System.out.println("Connecting to statsd at: " + this.statsdHost + ":" + this.statsdPort);
+        this.statsd = new NonBlockingStatsDClient(this.statsPrefix, this.statsdHost, this.statsdPort);
+    }
+
+    @Override
+    public void count(String stat, long delta){
+        statsd.count(stat, delta);
```

Comment:
But _this_ should be wrapped in a try-catch and perhaps logged back into the stdout logger to report on. We don't want issues in a third-party library affecting KafkaLog's ability to report on spraynozzle's performance. (Which also has no exception handling, but in this case would cause the thread to crash while leaving the rest running in a half-broken state where eventually the log queue would fill up and kill the application.)